src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -118,8 +154,16 @@ internal static class PeWriter
             ushort portablePdbVersion = 0;
             var metadataRootBuilder = mdWriter.GetRootBuilder();
 
+            Machine SRMVisibleMachine = properties.Machine;
+
+            // When attempting to write an Arm64 PE file, tell the PEBuilder that an Amd64 pe is being written instead
+            // then replace the bits in the produced PE header with the correct bits. This will allow an Arm64 pe to be
+            // generated without requiring the System.Reflection.Metadata dll to be updated to a newer version.
```

Comment:
>System.Reflection.Metadata dll [](start = 47, length = 30)

Is there a bug on `System.Reflection.Metadata dll` to start supporting this?
I think the hacks are ok for now, but should be commented with a reference to the bug so that when bug is fixed we do not forget to go and fix that up.

Reply:
System.Reflection.Metadata does support this, but we can't yet take an update to use it.

Reply:
We just need an easily identifiable comment referring to a tracking bug, so that when we get new S.R.M. it would be trivial to remove, then unnecessary, workarounds